### PR TITLE
Fix updating objects with null values

### PR DESCRIPTION
**Pull request by dirkkul**
### What's being changed:

Closes https://github.com/weaviate/weaviate/issues/2607

- First two commits add more acceptance tests around schema casing. Both worked without any further changes.
- When sending an update with an explicit null value it was ignored. Now it is removed from the object
- Adds an acceptance test using the go client

### Review checklist

- [x] All new code is covered by tests where it is reasonable.

### CODE REVIEW

File: test/acceptance_with_go_client/autoschema_test.go [hunk 1]
```diff
@@ -77,13 +75,73 @@ func TestAutoschemaCasingProps(t *testing.T) {
...
+func TestAutoschemaCasingUpdateProps(t *testing.T) {
+	ctx := context.Background()
+	c := client.New(client.Config{Scheme: "http", Host: "localhost:8080"})
+
+	objId := "67b79643-cf8b-4b22-b206-6e63dbb4e57a"
+	upperPropName := "SomeProp"
+	lowerPropName := "someProp"
+	cases := []struct {
+		prop1 string
+		prop2 string
```
Comment (aliszka) [inline test/acceptance_with_go_client/autoschema_test.go:114]: what is `prop2` for? seems to be used just as test name

File: test/acceptance_with_go_client/schema_test.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		{className1: lowerClassName, className2: upperClassName},
+	}
+	for _, tt := range cases {
+		t.Run(tt.className1+" "+tt.className2, func(t *testing.T) {
+			c.Schema().ClassDeleter().WithClassName(tt.className1).Do(ctx)
+			c.Schema().ClassDeleter().WithClassName(tt.className2).Do(ctx)
+			class := &models.Class{Class: tt.className1, Vectorizer: "none"}
+			require.Nil(t, c.Schema().ClassCreator().WithClass(class).Do(ctx))
+
+			// try to create class again lowercase. This should fail as it already exists
```
Comment (aliszka) [inline test/acceptance_with_go_client/schema_test.go:49]: `lowercase` -> depends on test case currently tested

File: test/acceptance_with_go_client/schema_test.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+			class2 := &models.Class{Class: tt.className2, Vectorizer: "none"}
+			require.NotNil(t, c.Schema().ClassCreator().WithClass(class2).Do(ctx))
+
+			// create object with lower and upper case class names. Both should succeed
```
Comment (aliszka) [inline test/acceptance_with_go_client/schema_test.go:53]: `lower and upper` -> depends on test case currently tested

**Review by aliszka — APPROVED** [review]

File: test/acceptance_with_go_client/autoschema_test.go (hunk 1 above)
Comment (dirkkul) [reply test/acceptance_with_go_client/autoschema_test.go:114]: whoopts, that should have been the name for the update, fixed

File: test/acceptance_with_go_client/schema_test.go (hunk 2 above)
Comment (dirkkul) [reply test/acceptance_with_go_client/schema_test.go]: also fixed :)

**Review by antas-marcin — APPROVED** [review]

**Pull request merged**